test/whitequark/test_assignment_to_numparams_0.rb

```diff
@@ -0,0 +1,4 @@
+# typed: true
+
+proc {_1 = nil}
+    # ^^ error: cannot assign to numbered parameter _1
```

Comment:
Are there situations where it's ok to assign to `_1` as a local variable? Can we write tests for those?

like is this valid:

```ruby
_1 = true
puts _1
```

?

Reply:
Yes this is accepted by the syntax (I added a test) but should it?

Whitequark allows it but Ruby 2.7 gives a `warning: '_1' is reserved for numbered parameter; consider another name`.

This also made me realize that Whitequark will error for any assignment to a numparam inside a block while Ruby will only warn. Should we remove the error?

Reply:
I'm fine with the error. We can relax the error if people complain. Thanks for having tests!